tests/e2e-kubernetes/testsuites/credentials.go

```diff
@@ -903,6 +1073,19 @@ func oidcProviderForCluster(ctx context.Context, f *framework.Framework) string
 	return strings.TrimPrefix(issuer, "https://")
 }
 
+//-- EKS Pod Identity utils
+
+// eksPodIdentityAgentDaemonSetForCluster tries to find configured EKS Pod Identity Agent for the cluster we're testing against.
```

Comment:
A bit confused on how do we install EKS Pod Identity Agent in the first place?

Reply:
It's the first step of the cluster configuration done by the cluster administrator. It's usually [installed as an EKS-Addon](https://docs.aws.amazon.com/eks/latest/userguide/pod-id-agent-setup.html).

Reply:
For e2e tests we create and configure EKS clusters ourselves in https://github.com/awslabs/mountpoint-s3-csi-driver/blob/main/tests/e2e-kubernetes/scripts/eksctl.sh#L50.

You would either need to pre-install it during cluster creation script (similarly to CloudWatch Addon https://github.com/awslabs/mountpoint-s3-csi-driver/blob/main/tests/e2e-kubernetes/scripts/eksctl-patch.json#L33-L39)
 or install/uninstall it dynamically during test runs.